src/main/java/org/apache/sling/cli/impl/http/HttpClientFactory.java

```diff
@@ -61,9 +67,18 @@ protected void activate(ComponentContext ctx) {
     }
 
     public CloseableHttpClient newClient() {
-        
+        final AtomicReference<String> url = new AtomicReference<>();
```

Comment:
Do we need safe publication or is this just for convenience? If convenience, I suggest using a 1-element string array as I think it's clearer.

Reply:
I usually use this pattern when having to deal with outside context in a lambda. We could also switch to your proposal, if you think it makes more sense.